images/Dockerfile

```diff
@@ -0,0 +1,54 @@
+#
+# BSD 2-Clause License
+#
+# Copyright (c) 2021, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+## Builder image
+FROM ubuntu:21.04 as builder
+
+# Install tooling
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -q -y --no-install-recommends \
+    ca-certificates git \
+    make cmake gcc g++ libc6-dev python3 libpython3-dev pip && \
+    rm -rf /var/lib/apt/lists/*
+RUN ln -s /usr/bin/python3 /usr/bin/python
+RUN pip3 install cmake
+
+# Get source code
+WORKDIR /usr/local/src/
+RUN git config --global advice.detachedHead false && \
+    git clone https://github.com/CrayLabs/SmartRedis.git
```

Comment:
``--branch``?

Reply:
Yeah, this brings up a good point.  By default this will clone from ``develop`` which is not what we want for the github actions.  I attempted to get a ``--branch`` option to work with github actions via docker ``--build-args``, but this ended up being problematic with the way that github actions does the branch and sha names differently for ``push`` actions and ``pull request`` actions.  As a result, I've left a comment in the Dockerfile for the user (with code) on how to clone a specific repository and branch for use in the Dockerfile, but otherwise the default behavior will be to build from the current state of the repo in which ``./images/build.sh`` is invoked.  This guarantees that our github actions are executed correctly for both push and pull request actions.  After doing some research, I have seen this as accepted behavior (and often preferred) so that developers can make and test changes to the project without committing and pushing changes.  I'll make sure to document this in the doc updates associated with this PR.